### ISSUE TITLE
Account for assistant when centering feature

### ISSUE THREAD

**Issue by quincylvania** [openstreetmap/iD#6657]
The assistant can sometimes cover part of the feature.

![Screen Shot 2019-07-17 at 1 33 25 PM](https://user-images.githubusercontent.com/2046746/61397271-8536cd00-a897-11e9-8b47-2734272bc365.png)

**Comment by bhousel** [comment]
This used to work pretty well when it was the inspector.  Would you consider bringing the inspector back?

**Comment by quincylvania** [comment]
> Would you consider bringing the inspector back?

We can have a broader discussion about this if you want, but for this particular issue we just need to account for the offset. iD already accounts for the issues pane if it's open when you zoom to issues, so we can adopt that code.

**Comment by bhousel** [comment]
I am really struggling with this assistant concept.
- we can't hover over features to inspect them anymore
- map tiles which end up mostly invisible are fetched/drawn unnecessarily underneath it
- the scale bar is now in a position where you can't actually measure things
- #6768 where UI looks odd for controls that expect the padding that the inspector provided

It's not clear to me what actual problem this change was supposed to solve.  The inspector was good in a lot of ways.

**Comment by quincylvania** [comment]
> The inspector was good in a lot of ways.

Agreed! My goal for the assistant is that it's just as good but less obtrusive. It still needs more work to get there, and I'm open to feedback.

> It's not clear to me what actual problem this change was supposed to solve.

The problem I see with the the sidebar is that it always takes up the same amount of space even when the space isn't used. This might not be an issue on large displays, but on laptops, tablets, and phones this makes a big difference. The assistant is never bigger than it needs to be for its content.

The sidebar also constrains the width of the top and bottom toolbars. With the assistant, these always stay full-width.

<img width="1280" alt="Screen Shot 2019-08-19 at 9 49 10 AM" src="https://user-images.githubusercontent.com/2046746/63275840-dd326c00-c267-11e9-846e-8c6c16e3f3a0.png">

<img width="1280" alt="Screen Shot 2019-08-19 at 9 54 39 AM" src="https://user-images.githubusercontent.com/2046746/63276496-1a4b2e00-c269-11e9-8e07-f88821653451.png">
<img width="1280" alt="Screen Shot 2019-08-19 at 9 54 47 AM" src="https://user-images.githubusercontent.com/2046746/63276503-1fa87880-c269-11e9-84ec-03147e9cfb8a.png">

The issues you mentioned are real, but I don't see them as unsolvable with this design.

> - we can't hover over features to inspect them anymore

I've been planning on adding this feature back.

> - map tiles which end up mostly invisible are fetched/drawn unnecessarily underneath it

True, but it's not a lot of area and the user could scroll over or collapse the assistant at any time, so I don't see this as much of a waste.

> - the scale bar is now in a position where you can't actually measure things

We can move it to the opposite side of the window.

> - #6768 where UI looks odd for controls that expect the padding that the inspector provided

I'm not exactly sure what you mean here, but we can add more padding within the assistant.

**Comment by BjornRasmussen** [comment]
I like the assistant because it frees up space on the map, but do miss the hover to view fields feature and old preset switching system.

**Comment by k-yle** [comment]
I'm going to close this issue, since it's an issue in the v3 prototype version, which is (sadly) abandoned and [no longer deployed](https://github.com/openstreetmap/iD/commit/36f4eeb7e24ba3ffd16f0d48780f3c56d39f2bd1).
